aws/data_source_aws_glue_data_catalog_encryption_settings.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAwsGlueDataCatalogEncryptionSettings() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceAwsGlueDataCatalogEncryptionSettingsRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
```

Comment:
id attribute is implicit dont think we need this.

Reply:
Is it also implicitly required? The goal here is to make `id` a required argument, so just want to make sure that behavior would remain the same if I take this out.

Reply:
You are right, my mistake i see this is the convention in data sources to keep "id" explicit. ill try to run tests soon. code wise looks good